Apps/AdminWebClient/src/Server/Services/InactiveUserService.cs

```diff
@@ -0,0 +1,195 @@
+                }
+                else
+                {
+                    requestResult.ResultStatus = ResultType.Error;
+                    requestResult.ResultError = adminUsersResult.ResultError;
+                }
+
+                if (supportUsersResult.ResultStatus == ResultType.Success)
+                {
+                    List<UserRepresentation> supportUsers = supportUsersResult.ResourcePayload.ToList();
+                    this.UpdateInactiveUser(inactiveUsers, supportUsers, IdentityAccessRole.SupportUser);
+                    this.AddInactiveUser(inactiveUsers, activeUserProfiles, supportUsers, IdentityAccessRole.SupportUser);
+                }
+                else
+                {
+                    requestResult.ResultStatus = ResultType.Error;
+                    requestResult.ResultError = supportUsersResult.ResultError;
+                }
+
+                if (requestResult.ResultStatus == ResultType.Success)
+                {
+                    this.logger.LogDebug("Inactive user with no keycloak match count: {Count}...", inactiveUsers.FindAll(x => x.UserId == null).Count);
+                    inactiveUsers.RemoveAll(x => x.UserId == null);
+                    requestResult.ResourcePayload = inactiveUsers;
+                    requestResult.TotalResultCount = inactiveUsers.Count;
+                }
+            }
+        }
+
+        this.logger.LogDebug("Inactive user count: {Count}...", inactiveUsers.Count);
+        return requestResult;
+    }
+
+    private void UpdateInactiveUser(List<AdminUserProfileView> inactiveUsers, List<UserRepresentation> identityAccessUsers, IdentityAccessRole role)
+    {
+        this.logger.LogDebug("{Role} count: {Count}...", role.ToString(), identityAccessUsers.Count);
+        List<UserRepresentation> users = identityAccessUsers.FindAll(x1 => inactiveUsers.Exists(x2 => x1.Username == x2.Username));
+        this.logger.LogDebug("Filtered matching {Role} count: {Count}...", role.ToString(), users.Count);
+
+        foreach (AdminUserProfileView inactiveUser in inactiveUsers)
+        {
+            UserRepresentation? user = users.Find(x => x.Username == inactiveUser.Username);
+
+            if (user != null)
+            {
+                inactiveUser.UserId = inactiveUser.UserId ??= user.UserId;
+                inactiveUser.Email = inactiveUser.Email ??= user.Email;
+                inactiveUser.FirstName = inactiveUser.FirstName ??= user.FirstName;
+                inactiveUser.LastName = inactiveUser.LastName ??= user.LastName;
+                inactiveUser.RealmRoles = inactiveUser.RealmRoles != null ? (inactiveUser.RealmRoles + ", " + role.ToString()) : role.ToString();
+            }
+        }
+    }
+
+    private void AddInactiveUser(List<AdminUserProfileView> inactiveUsers, List<AdminUserProfile> activeUserProfiles, List<UserRepresentation> identityAccessUsers, IdentityAccessRole role)
```

Comment:
Nit: would suggest renaming to `AddInactiveUsers`.

Reply:
Addressed.